smack-core/src/main/java/org/jivesoftware/smack/util/ParserUtils.java

```diff
@@ -194,4 +198,17 @@ public static URI getUriFromNextText(XmlPullParser parser) throws XmlPullParserE
         return uri;
     }
 
+    public static Map<String, String> getCustomAttributes(XmlPullParser parser) {
+        Map<String, String> customAttributes = new LinkedHashMap<>();
+        List<String> standardAttributes = Arrays.asList(new String[] {"xml:lang", "id", "to", "from", "type"});
```

Comment:
Why construct this every time the method is called? And why is it a `List`? This should be a static `Set`.